### PR TITLE
Feature/use git info cache for all solution projects

### LINKED ISSUE #190: make the gitinfo caches location configurable to be able to reuse the same information over several projects that are built
i would like to use GitInfo to version all DLLs in my solution to the same version. i use SDK style projects and would like to use the generated assembly. therefore I placed a central Build.Directory.targets for the solution that would add the following:
```
    <ItemGroup>
        <PackageReference Include="GitInfo" Version="2.2.0">
            <PrivateAssets>all</PrivateAssets>
            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
        </PackageReference>
    </ItemGroup>

    <Target Name="SetVersionFromGitInfo" DependsOnTargets="GitVersion" BeforeTargets="GetAssemblyVersion">
        <PropertyGroup>
            <BaseVersion>$(GitBaseVersionMajor).$(GitBaseVersionMinor).$(GitBaseVersionPatch).$(GitCommits)</BaseVersion>

            <!-- for TFS build GitBranch is not set so we need to get the branch from the TFS variable -->
            <FeatureBranch>$(BUILD_SOURCEBRANCH)</FeatureBranch>
            <FeatureBranch Condition="$(FeatureBranch.StartsWith('refs/heads/'))">$(FeatureBranch.Substring(11))</FeatureBranch>
            <!-- if variable is not set its a local build -->
            <FeatureBranch Condition="$(FeatureBranch) == ''">$(GitBranch)</FeatureBranch>

            <ExtendedVersion>$(BaseVersion) / $(GitCommit) ($(GitIsDirty)) $(FeatureBranch)</ExtendedVersion>

            <Version>$(BaseVersion)</Version>
            <InformationalVersion>$(ExtendedVersion)</InformationalVersion>
        </PropertyGroup>
    </Target>
```
i recognized that the build time for the round 30 projects increased about 2 minutes after that change.
so i tried to see how to optimize that a bit. we found that the GitInfo for a single project would be cached so we thought it might be a nice approach to move the cache to the solution directory and reuse it so that not every project is recreating its own cache with the same content.

to test that i added the following to the central .targets file: 
```
    <Target Name="OverrideGitInfoCache" BeforeTargets="$(GitInfoDependsOn)">
        <PropertyGroup>
            <_GitInfoFile>../GitInfo.cache</_GitInfoFile>
            <_GitIsDirtyFile>../GitIsDirty.cache</_GitIsDirtyFile>
        </PropertyGroup>
    </Target>
```

it almost worked. i just recognised that now the cache is still updated on each project, since the is dirty cache was updated all the time and therefore the gitinfo cache was recreated. i think that could be fixed with the change i commented in #165.

it would be nice to be able to change the cache location by just setting a variable. maybe this could be something that should be configurable.

### PR DESCRIPTION
**Pull request by tisis2**
closes #190 

use $(GitCachePath) to overwrite the cache location. the GitIsDirty.cache should now be overwritten.

### CODE REVIEW

File: src/GitInfo/build/GitInfo.targets [hunk 1]
```diff
@@ -339,7 +347,7 @@
 			<GitIsDirty>0</GitIsDirty>
 		</PropertyGroup>
 
-		<WriteLinesToFile File="$(_GitIsDirtyFile)" Lines="$(GitIsDirty)" WriteOnlyWhenDifferent="true"/>
+		<WriteLinesToFile File="$(_GitIsDirtyFile)" Lines="$(GitIsDirty)" WriteOnlyWhenDifferent="true" Overwrite="true"/>
```
Comment (DanielTheCoder) [inline src/GitInfo/build/GitInfo.targets:350]: Overwrite also be solved here: https://github.com/devlooped/GitInfo/pull/184

Comment (kzu) [reply src/GitInfo/build/GitInfo.targets:350]: That PR has been merged now. Hopefully the update will be trivial for this one.

**Comment by kzu** [comment]
Ok, should be ready to merge as soon as the CLA is signed and conflicts resolved. Sorry for the delay!

**Comment by tisis2** [comment]
resolved conflicts and signed cla. so just waiting for the merge

**Comment by kzu** [comment]
ok, for whatever reason, it's not allowing me to rebase your branch, so you'll have to do it again I'm afraid :(

**Comment by kzu** [comment]
Dang, merged more PRs and now it's again out of sync :(. Sorry! Next release is yours alone 😉

**Pull request merged**